### PR TITLE
fix(ext/node): initial  `crypto.createPublicKey()` support

### LINKED ISSUE #21807: Implement `crypto.createPublicKey()`

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #21807

CC @littledivy

### CODE REVIEW

File: ext/node/ops/crypto/mod.rs [hunk 1]
```diff
@@ -1455,3 +1456,102 @@ pub fn op_node_create_private_key(
...
+
+#[op2]
+#[serde]
+pub fn op_node_create_public_key(
+  #[buffer] key: &[u8],
+  #[string] format: &str,
+  #[string] type_: &str,
+) -> Result<AsymmetricKeyDetails, AnyError> {
+  let doc = parse_public_key(key, format, type_)?;
+  let pk_info = pkcs8::SubjectPublicKeyInfo::try_from(doc.as_bytes())?;
```
Comment (iuioiua) [inline ext/node/ops/crypto/mod.rs:1488]: Rust is telling me the value I'm passing to `try_from()` is incorrect.

File: ext/node/ops/crypto/mod.rs [hunk 2]
```diff
@@ -1455,3 +1456,102 @@ pub fn op_node_create_private_key(
...
+        modulus_length,
+        public_exponent: BigInt::from_bytes_be(
+          num_bigint::Sign::Plus,
+          public_key.public_exponent.as_bytes(),
+        )
+        .into(),
+      })
+    }
+    RSASSA_PSS_OID => {
+      let params = PssPrivateKeyParameters::try_from(
```
Comment (iuioiua) [inline ext/node/ops/crypto/mod.rs:1508]: IIUC, the `PssPrivateKeyParameters` should be renamed to `PssKeyParameters`, as it also applies to public keys.

**Comment by iuioiua** [comment]
Great stuff! LGTM. However, this should be marked as "feat".

**Comment by littledivy** [comment]
@iuioiua All the Node API changes are merged as fixes so we can include them in patch releases.

**Review by littledivy — APPROVED** [review]

**Pull request merged**